API changes for Kubernetes version 2.0.0

brejoc opened the pull request:
### What does this PR do?

~~Switching to the new [Kubernetes client lib](https://pypi.python.org/pypi/kubernetes/2.0.0) API introduced with version 2.0.0.~~
Made Saltstack Kubernetes module version agnostic to Kubernetes Python client.

### What issues does this PR fix or reference?

#42843

### Previous Behavior
You had to use Kubernetes Python client version 1.0.2.

### New Behavior
Now you can use Kubernetes Python client version 2.0.0 and below.
### Tests written?

No

brejoc committed:
API changes for Kubernetes version 2.0.0

Switching to the new Kubernetes client lib API introduced with version 2.0.0.

salt-jenkins commented:
@brejoc, thanks for your PR! By analyzing the history of the files in this pull request, we identified @flavio, @meaksh and @gtmanfred to be potential reviewers.

gtmanfred reviewed (changes requested):
Is there a reason we can't gate these changes, and allow both the old and this newer version of kubernetes python client to be installed?

brejoc committed:
Not depending on specific K8s version anymore

Kubernetes imports are no longer version specific. Seems like the API change
was just a rename, so now we are importing either `V1beta1Deployment` or
`AppsV1beta1Deployment`. Usage stays the same.

brejoc commented:
@gtmanfred You are absolutely right – I think there is no reason not to do that. Thanks! Changed it.

brejoc commented:
@meaksh Would you be so kind to also take a look!? Thanks!

meaksh commented on salt/modules/kubernetes.py:
```diff
@@ -2,7 +2,7 @@
 Module for handling kubernetes calls.
 
-:optdepends:    - kubernetes Python client
+:optdepends:    - kubernetes Python client up to version 2.0.0
```
I would remove this `:optdepends:` change, since the current status of this PR also allows to use lower versions of kubernetes Python client.

meaksh reviewed (approved):
Apart from the previous comment about `:optdepends:` documentation tag, it looks good to me!

Thanks @brejoc for fixing this! 👍 😄

brejoc committed:
Version info in :optdepends: not needed anymore

Since this should work with 1.x.x and 2.x.x we don't have to provide a ersion info.

flavio commented:
LGTM too
